lib/streamlit/commands/navigation.py

```diff
@@ -30,12 +31,35 @@
 )
 
 if TYPE_CHECKING:
-    from streamlit.navigation.page import StreamlitPage
     from streamlit.source_util import PageHash, PageInfo
 
 SectionHeader: TypeAlias = str
 
 
+def convert_to_streamlit_page(
+    page_input: str | Callable[[], None] | StreamlitPage,
+) -> StreamlitPage:
+    """Convert various input types to StreamlitPage objects."""
+    if isinstance(page_input, StreamlitPage):
+        return page_input
+
+    if isinstance(page_input, str):
+        # Convert string path to StreamlitPage
+        return Page(Path(page_input))
+
+    if isinstance(page_input, Path):
+        return Page(page_input)
+
+    if callable(page_input):
+        # Convert function to StreamlitPage
+        return Page(page_input)
```

Comment:
For these, we want to use `StreamlitPage`, not `Page`. Page is annotated to assume it's an st call where StreamlitPage is the underlying function.